lib/utils/narrowFixRange.mjs

```diff
@@ -0,0 +1,68 @@
+/** @import { Node as PostcssNode } from 'postcss' */
+
+/**
+ * Make a fix range as narrow as possible by trimming common prefixes and suffixes
+ *
+ * @param {PostcssNode} node
+ * @param {{range: [number, number], text: string}} fixData
+ * @returns {{range: [number, number], text: string}}
+ */
+export default function narrowFixRange(node, fixData) {
+	if (!node.source) {
+		return fixData;
+	}
+
+	let startOffset = fixData.range[0];
+	let endOffset = fixData.range[1];
+
+	const stringRepresentation = node.source.input.css.slice(startOffset, endOffset);
+
+	let replacementStartOffset = 0;
+
+	for (let i = 0; i < stringRepresentation.length; i++) {
+		const a = stringRepresentation[i];
+		const b = fixData.text[i];
+
+		if (a !== b) break;
+
+		startOffset++;
+		replacementStartOffset++;
+	}
+
+	let replacementEndOffset = fixData.text.length;
+
+	for (let i = 0; i < stringRepresentation.length; i++) {
+		const ia = stringRepresentation.length - 1 - i;
+
+		if (ia < replacementStartOffset) break;
+
+		const a = stringRepresentation[ia];
+
+		const ib = fixData.text.length - 1 - i;
+
+		if (ib < replacementStartOffset) break;
+
+		const b = fixData.text[ib];
+
+		if (a !== b) break;
+
+		endOffset--;
+		replacementEndOffset--;
+	}
+
+	// Try to have a context size of at least 1
+	if (startOffset === endOffset) {
+		if (startOffset > fixData.range[0]) {
+			startOffset--;
+			replacementStartOffset--;
+		} else if (endOffset < fixData.range[1]) {
+			endOffset++;
+			replacementEndOffset++;
+		}
+	}
```

Comment:
I've noticed that the width of a range can be 0 columns. This happens on insertions. (e.g. append `;` at the end of a declaration).

An insert in the middle of a PostCSS node is unambiguous. In that case it is always clear which node should be skipped on subsequent fix attempts.

But an insert at the start or end of a node is unclear.
In such cases it might be the start of the given node, or the end of the previous node.

By widening the fix range a tiny bit we have an unambiguous range.

Reply:
> By widening the fix range a tiny bit we have an unambiguous range.

Iv reached these tests.
i.e. `zz` instead of `z`

I was wondering, if the first character of the next node is the same as the penultimate character of the previous node, do we have a problem?
i.e. the "context size" is too small
If we do, should we increase the number of characters checked until it differs?
i.e. 1, 2, etc. until length
Maybe it never happens or I misunderstood completely, if so you can ignore this.

Reply:
The ambiguity is not one of characters being the same. It is purely mathematical.
I am not sure how to better explain it both in words and in test cases :)